internals/overlord/logstate/opentelemetry/opentelemetry.go

```diff
@@ -264,37 +253,34 @@ func (c *Client) Flush(ctx context.Context) error {
 	// Sort service names to ensure deterministic order.
 	sort.Strings(serviceNames)
 
-	resourceLogs := make([]ResourceLogs, 0, len(serviceNames))
+	logs := make([]resourceLogs, 0, len(serviceNames))
 	for _, serviceName := range serviceNames {
 		batch := serviceBatches[serviceName]
 		if len(batch) > 0 {
```

Comment:
Nit: it would avoid an extra nesting level to use an early-continue here: `if len(batch) == 0 { continue }`